doc/specs/bookmarks.md

```diff
@@ -0,0 +1,595 @@
+ - The Request Name/Year, Filled by, and Requested by columns should have fields that are hyperlinked to their respective pages.
+ 
+
+#### 4.1 Adding a request Bookmark
+
+**Primary Actor:** Any user.
+
+**Brief:** Allows a user to add a bookmark for a request.
+
+**Trigger:** The user wishes to bookmark a request.
+
+**Preconditions:** The user has not already bookmarked the request
+
+**Postconditions:** The request is now bookmarked for that user
+
+**Basic Flow:**
+
+ 1. The user navigates to the request page.
+ 2. The user clicks the "Bookmark" hyperlink.
+ 3. The request is bookmarked for that user.
+ 
+**Additional functional requirements:**
+ - The hyperlink function should also change to that of removing the bookmark
+
+**Additional non-functional requirements:** 
+ -  The hyperlink text is updated to "Remove Bookmark"
+ 
+ 
+#### 4.2 Removing a request Bookmark
+
+**Primary Actor:** Any user.
+
+**Brief:** Allows a user to remove a bookmark for a request.
+
+**Trigger:** The user wishes to remove the bookmark for a request.
+
+**Preconditions:** The user has already bookmarked the request.
+
+**Postconditions:** The request is no longer bookmarked for that user.
+
+**Basic Flow:** 
+
+ 1. The user navigates to the request page.
+ 2. The user clicks the "Remove Bookmark" hyperlink.
+ 3. The request bookmark is removed.
+
+**Additional functional requirements:** 
+ - The hyperlink function should also change to that of adding a bookmark.
+
+**Additional non-functional requirements:**
+ - The hyperlink text is updated to "Bookmark" 
+ 
+ **Alternate Flow:**
+ 
+ 1. The user has navigated to the Bookmarks page for requests.
+ 2. The user clicks the "Remove Bookmark" hyperlink.
+ 3. The request bookmark is removed.
+ 
+**Additional non-functional requirements:**
+ - The request row should be removed from the page. 
+ 
+ 
+#### 4.3 Sorting All Bookmarked Request Rows 
+
+**Primary Actor:** Any user.
+
+**Brief:** Allows the user to sort the rows of a bookmark collection based on different fields.
+
+**Trigger:** The user wants to sort by Year, Votes, Bounty, Filled, Requested by, 
+Created, or Last Vote
+
+**Preconditions:** The user has already bookmarked at least two collages. 
+
+**Basic Flow:**
+ 1. The user navigates to the Bookmarks page for collages.
+ 2. The user clicks on the column title.
+
+**Additional functional requirements:**
+ - If a column is not currently being used for sorting all other column sorts are removed and 
+ it is the table is sorted in descending order by that column.
+ - If the column is currently used for descending order, the order is now set to 
+ ascending order for that column.
+ - If the column is currently used for ascending order, the order is now set to 
+ descending order for that column.
+ 
+**Additional non-functional requirements:** 
+ - The table order should be sorted based on the user selection and logic above. 
+ 
+ 
+#### 4.4 Remove Filled Bookmarked Requests
+ 
+**Primary Actor:** Any user.
```

Comment:
This should be "A user"
"Any user" implies that **I** can remove **your** filled bookmarked requests